marlowe-playground-client/src/HaskellEditor/View.purs

```diff
@@ -87,35 +86,48 @@ haskellEditor ::
   ComponentHTML Action ChildSlots m
 haskellEditor state = slot _haskellEditorSlot unit component unit (Just <<< HandleEditorMessage)
   where
-  setup editor =
-    liftEffect do
-      -- TODO we shouldn't access local storage from the view
-      mContents <- LocalStorage.getItem StaticData.bufferLocalStorageKey
-      let
-        contents = fromMaybe HE.escrow mContents
-      model <- Monaco.getModel editor
-      Monaco.setValue model contents
+  setup editor = pure unit
 
   component = monacoComponent $ HM.settings setup
 
 compileButton :: forall p. State -> HTML p Action
 compileButton state =
-  button [ onClick $ const $ Just Compile ]
-    [ text (if has (_compilationResult <<< _Loading) state then "Compiling..." else "Compile") ]
-
-sendResultButton :: forall p. State -> String -> Action -> HTML p Action
-sendResultButton state msg action =
-  let
-    compilationResult = view _compilationResult state
-  in
-    case view _compilationResult state of
-      Success (Right (InterpreterResult result)) ->
-        button
-          [ onClick $ const $ Just action
-          , disabled (isLoading compilationResult || (not isSuccess) compilationResult)
-          ]
-          [ text msg ]
-      _ -> text ""
+  button
+    [ onClick $ const $ Just Compile
+    , enabled enabled'
+    , classes classes'
+    ]
+    [ text buttonText ]
+  where
+  buttonText = case view _compilationResult state of
+    Loading -> "Compiling..."
+    (Success _) -> "Compiled"
```

Comment:
I think the top-level parenthesis around patterns in cases are not necessary in general

Reply:
changed 👍🏻